UnitsHelper source/header requires additional headers with libstdc++ version 13.1.0

`libstdc++` (GNU C++) version 13 raises issues with string comparison, and std::runtime_error not being found, when compiling target `core_mediator`. `libc++` (LLVM C++) compiles as expected.

## Current behavior
```
...
[build] ngen/src/core/mediator/UnitsHelper.cpp:42:24: error: no member named 'runtime_error' in namespace 'std'
[build]             throw std::runtime_error("Unable to convert " + in_units + " to " + out_units);
[build]                   ~~~~~^
[build] ngen/src/core/mediator/UnitsHelper.cpp:60:17: error: invalid operands to binary expression ('const std::string' (aka 'const basic_string<char>') and 'const std::string')
[build]     if(in_units == out_units){
[build]        ~~~~~~~~ ^  ~~~~~~~~~
...
```

## Expected behavior
The `core_mediator` target should compile.

## Steps to replicate behavior (include URLs)

> Assumes CMake project is already configured, and the system default is to use `libstdc++` with **clang** or **icx**.

```sh
# This should fail
cmake --build build --target core_mediator
```

Add the following at around `CMakeLists.txt:47` to force `libc++`
```cmake
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
```
and rerun
```sh
# This should compile
cmake --build build --target core_mediator
```

## Possible Solution
Add an include directive for `<stdexcept>` (which also includes `<string>`) to `include/core/mediator/UnitsHelper.hpp`.

I'm guessing either `<map>`, `<memory>`, or `<mutex>` stopped using `<string>` and `<stdexcept>` so they aren't included by association anymore, but still are in `libc++`. (Unless this is only an issue in my environment...)

--- a/include/core/mediator/UnitsHelper.hpp
+++ b/include/core/mediator/UnitsHelper.hpp
@@ -2,6 +2,7 @@
 #include <map>
 #include <memory>
 #include <mutex>
+#include <stdexcept>
 #include "all.h"
 
 #ifndef NGEN_UNITSHELPER_H

